project-base/app/tests/FrontendApiBundle/Functional/Complaint/ComplaintTestTrait.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrontendApiBundle\Functional\Complaint;
+
+use Shopsys\FrameworkBundle\Model\Complaint\Complaint;
+use Shopsys\FrameworkBundle\Model\Complaint\ComplaintItem;
+use Shopsys\FrontendApiBundle\Model\ScalarType\DateTimeType;
+
+trait ComplaintTestTrait
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\ComplaintItem $expectedComplaintItem
+     * @param array $complaintItem
+     */
+    protected function assertComplaintItem(ComplaintItem $expectedComplaintItem, array $complaintItem): void
+    {
+        $complaintItemMessage = sprintf(
+            'Hint: check data of complaint item with ID #%d',
+            $expectedComplaintItem->getId(),
+        );
+
+        $this->assertArrayHasKey('quantity', $complaintItem, $complaintItemMessage);
+        $this->assertSame($expectedComplaintItem->getQuantity(), $complaintItem['quantity'], $complaintItemMessage);
+
+        $this->assertArrayHasKey('description', $complaintItem, $complaintItemMessage);
+        $this->assertSame($expectedComplaintItem->getDescription(), $complaintItem['description'], $complaintItemMessage);
+
+        $this->assertArrayHasKey('orderItem', $complaintItem, $complaintItemMessage);
+
+        $orderItem = $complaintItem['orderItem'];
+        $expectedOrderItem = $expectedComplaintItem->getOrderItem();
+
+        $this->assertArrayHasKey('uuid', $orderItem, $complaintItemMessage);
+        $this->assertSame($expectedOrderItem->getUuid(), $orderItem['uuid'], $complaintItemMessage);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Complaint\Complaint $expectedComplaint
+     * @param array $complaint
+     */
+    protected function assertComplaint(Complaint $expectedComplaint, array $complaint): void
```

Comment:
maybe `$actualComplaintData` is more descriptive?

Reply:
changed [fixup! get complaints query test](https://github.com/shopsys/shopsys/pull/3362/commits/0196828ec99ce8b131b88a9b1c47d83e90ca4a6f)